packages/hackmcx-api/src/controllers/captions.controller.js

```diff
@@ -34,4 +34,44 @@ export async function postCaptions(req, res){
         res.statusCode = 404;
         res.send({ error: e })
     }
-}
\ No newline at end of file
+}
+
+export async function rateCaption(req, res){
+    function err(message){
+        res.statusCode = 400;
+        res.send({error: message})
+    }
+
+    if (!req.body.rating){
+        err('"rating" missing from request body. Rating must be a number from 0 to 1.')
+        return;
+    }
+    let rating = parseFloat(req.body.rating);
+    if (rating.isNaN){
```

Comment:
This didn't catch NaN, I think syntax is `isNaN(rating)`

Reply:
isNaN has similar problems. We really wanted `Number.isNaN(rating)`